### ISSUE TITLE
csrftoken ecrase by kobocat on same domain

### ISSUE THREAD

**Issue by jdugh** [kobotoolbox/kpi#2775]
Hi,

## Description

I have a own server install (without Docker) on my server.
I run all Kobotoolbox on the same origin, with some prefix :
Enketo is on /enketo
Kobocat on /kc
and Kpi on /

I am updating all my kobotoolbox components on my server and I have a problem with the csrftoken.
When I am connected on kpi, I have a csrftoken, generate by kpi, 64 char.
If I load a Kobocat (iframe) page, like Gallery, or legacy download, Kobocat erase my csrftoken and replace it with 32 char.
So my csrftoken for kpi is no longer valid.

## Steps to Reproduce

1. kpi and kobocat on same domain
2. log to kpi and go to "data"
3. go to 'Gallery'
4. Go to 'Table'
4. In the console, we have error "Cookie not matched" from the "react_devtools_backend.js"

Do you know if it is a settings/configuration problem, or maybe because django on kpi and kobocat is not the same version ?
Kpi and kobocat can shared the same csrftoken ? 
On my kpi #2.018.48 and kobo #2.018.48, I don't have this problem.

Thanks a lot
Jonathan

**Comment by jnm** [comment]
Hi Jonathan,

It sounds like a similar problem to this: https://github.com/kobotoolbox/kobocat/issues/612
…except we haven't seen it because we aren't running everything on a single domain.

We'd be much obliged if you could figure out where Django sets the length of the CSRF token. With that information, we should be able to fix KoBoCAT to match KPI quickly.

**Comment by jdugh** [comment]
@jnm Thank's for your response.
Following your suggestion, I looked at the length of the cookie :

I saw in django venv, the `django/middleware/csrf.py` file containing the variable `CSRF_TOKEN_LENGTH` (django 2.2.7) or `CSRF_KEY_LENGTH` (django 1.8.19).
on Django 2.2, `CSRF_TOKEN_LENGTH = 64`
on django 1.8, `CSRF_KEY_LENGTH = 32`
Okey, I set CSRF_KEY_LENGTH to 64 for kobocat. Manually, I couldn't find any settings to override it :(
I also set the variable (in the settings of kobocat) `CSRF_COOKIE_HTTPONLY to False`. Otherwise, react_devtools_backend will not be able to access the cookie when overwritten.
Everything seems to be working properly.
I will come back to this issue if this solution does not fully correct.

The big blocking point is having to modify the csrf.py file in the venv ...

Thank you for all the kobotoolbox solution !
